src/dicts.jl

```diff
@@ -117,6 +117,58 @@ function Base.get(d::LMDBDict{K,V}, key, default) where {K,V}
     end
 end
 
+function Base.get!(d::LMDBDict{K,V}, key, default) where {K,V}
+    txn_dbi_do(d, readonly = true) do txn, dbi
+        mdb_key_ref = Ref(MDBValue(toref(convert(K,key))))
+        mdb_val_ref = Ref(MDBValue())
+        # Get value
+        ret = _mdb_get(txn.handle, dbi.handle, mdb_key_ref, mdb_val_ref)
+        if ret == MDB_NOTFOUND
+            d[key] = default
+            return default
+        elseif ret == Cint(0)
+            return convert(V, mdb_val_ref)
+        else
+            throw(LMDB.LMDBError(ret))
+        end
+    end
+end
+
+function Base.get(f::F, d::LMDBDict{K,V}, key) where {K,V,F<:Function}
+    txn_dbi_do(d, readonly = true) do txn, dbi
+        mdb_key_ref = Ref(MDBValue(toref(convert(K,key))))
+        mdb_val_ref = Ref(MDBValue())
+        # Get value
+        ret = _mdb_get(txn.handle, dbi.handle, mdb_key_ref, mdb_val_ref)
+        if ret == MDB_NOTFOUND
+            default = f()
+            return default
+        elseif ret == Cint(0)
+            return convert(V, mdb_val_ref)
+        else
+            throw(LMDB.LMDBError(ret))
+        end
+    end
+end
+
+function Base.get!(f::F, d::LMDBDict{K,V}, key) where {K,V,F<:Function}
```

Comment:
```suggestion
function Base.get!(f::Union{Function, Type}, d::LMDBDict{K,V}, key) where {K,V}
```

Reply:
IIUC, we need to keep the type parameter here to force specialization.

So we'd do something like `function Base.get!(f::F, d::LMDBDict{K,V}, key) where {K,V,F<:Union{Function, Type}}`.

Reply:
```suggestion
function Base.get!(f::F, d::LMDBDict{K,V}, key) where {K,V,F<:Union{Function, Type}}
```

Reply:
Yes, your suggestion looks good